opentelemetry-sdk/src/opentelemetry/sdk/util/__init__.py

```diff
@@ -13,9 +13,96 @@
 # limitations under the License.
 
 import datetime
+import logging
 import threading
 from collections import OrderedDict, deque
 from collections.abc import MutableMapping, Sequence
+from types import MappingProxyType
+from typing import MutableSequence
+
+from opentelemetry.util import types
+
+_VALID_ATTR_VALUE_TYPES = (bool, str, int, float)
+
+
+_logger = logging.getLogger(__name__)
+
+
```

Comment:
🤔 I guess other SDKs could use a method like this to ensure their attributes are all validated and users could potentially use this to validate/filter their attributes before setting them? not sure how likely these scenarios are though. I could go either ways with this.

Reply:
Are `Attributes`s an API concept? I am inclined towards keeping them in sdk because all operations are no-op in api package so we will probably never use them there and if we don't want offer them as a part of public api to user/application owners I don't see any benefit in moving to api package?

Reply:
Yes `Attribute` s are defined in the API. Hmm now thinking about it, this is probably valuable and common enough to be exposed in the public api. It is also defined in the specs so I don't mind increasing our api surface for this. For example, this would be useful for instrumentations to leverage (which only have dependencies on the api).

Reply:
I can move this to api util package. I just want to make sure we are okay with making these public because of all of util is pretty much private and internal.

Reply:
You can be the first :)